polygeist/include/mlir/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.h

```diff
@@ -435,24 +435,23 @@ class MemoryAccessAnalysis {
 
   MemoryAccessAnalysis(Operation *op, AnalysisManager &am);
 
+  /// Consumers of the analysis must call this member function immediately after
+  /// construction.
   MemoryAccessAnalysis &initialize(bool relaxedAliasing) {
-    this->relaxedAliasing = relaxedAliasing;
+    build(relaxedAliasing);
     return *this;
   }
 
   bool isInvalidated(const AnalysisManager::PreservedAnalyses &pa);
 
-  /// Return the operation this analysis was constructed from.
-  Operation *getOperation() const { return operation; }
-
   /// Return the memory access for the given memref \p access.
   std::optional<MemoryAccess>
   getMemoryAccess(const affine::MemRefAccess &access) const;
 
 private:
   /// Construct the access matrix and offset vector for the memory accesses
   /// contained in the operation associated with the analysis.
-  void build();
+  void build(bool relaxedAliasing);
```

Comment:
Nit: Do we still need both `initialize` and `build`? Seems like the only call to `build` is from `initialize`, so might as well inline that there.

Reply:
Yeah initialize may as well build.

Reply:
done